### ISSUE TITLE
Zeep fails trying to instantiate a client in python3.12

### ISSUE THREAD

**Issue by sdishman-bl** [mvantellingen/python-zeep#1412]
I suspect this is already known, as python3.12 is not listed on the supported versions on the main page.  But I also couldn't find a bug specifically calling it out.

1. The version of zeep (or if you are running master the commit hash/date): **4.2.1**
2. The WSDL you are using: I think any will do, but here's a public WSDL that I used to illustrate the error: **http://www.dneonline.com/calculator.asmx?WSDL**
3. And most importantly, a [runnable example script](http://docs.python-zeep.org/en/master/reporting_bugs.html) which exposes the problem.:

```
from zeep import Client
Client('http://www.dneonline.com/calculator.asmx?WSDL')
```

I saw that [this PR](https://github.com/mvantellingen/python-zeep/pull/1408) was submitted in February to address the issue, so I just wanted to highlight that here.

FWIW, we're in the middle of trying to update our systems to python3.12, and this happens to be the only known library preventing us from doing that.  I only offer that as "FYI", not as an attempt at escalation - we appreciate the work you do to make this library possible!

**Comment by davidve0206** [comment]
Hi, just a message to support this implementation - our system uses python 3.12 and we need to start using Zeep but are unable due to the version restrictions.

**Comment by AndrewMagerman** [comment]
+1 from us as well - we use heavily zeep, Support for python 3.12 would be hugely appreciated.

**Comment by h5rdly** [comment]
What needs to be done to add Python 3.12 support? Cheers

**Comment by pdewacht** [comment]
This crash might be caused by an outdated version of attrs. I can reproduce this with attrs 21.4.0 but not with later versions.

**Comment by mvantellingen** [comment]
Should be solved in the next version (4.3.0)

**Comment by sdishman-bl** [comment]
Thank you @mvantellingen !
